src/crypto-api/CryptoEvent.ts

```diff
@@ -0,0 +1,86 @@
+     * @param {boolean} status - true if the key backup is enabled, false otherwise.
+     */
+    KeyBackupStatus = "crypto.keyBackupStatus",
+    /**
+     * Fires when we failed to back up the keys
+     * @param {string} errorCode - the error code of the error that occurred
+     */
+    KeyBackupFailed = "crypto.keyBackupFailed",
+    /**
+     * Fires when the number of sessions that can be backed up changes.
+     * @param {number} remaining - the remaining number of sessions that can be backed up.
+     */
+    KeyBackupSessionsRemaining = "crypto.keyBackupSessionsRemaining",
+    /**
+     * Fires when a new valid backup decryption key is in cache.
+     * This will happen when a secret is received from another session, from secret storage,
+     * or when a new backup is created from this session.
+     *
+     * The payload is the version of the backup for which we have the key for.
+     *
+     * This event is only fired by the rust crypto backend.
+     */
+    KeyBackupDecryptionKeyCached = "crypto.keyBackupDecryptionKeyCached",
+    /**
+     * Fires when a key verification request is received.
+     * @param {VerificationRequest} - the request that was received
+     */
+    VerificationRequestReceived = "crypto.verificationRequestReceived",
+    /** @deprecated Use {@link DevicesUpdated} instead when using rust crypto */
+    WillUpdateDevices = "crypto.willUpdateDevices",
+    /**
+     * Fires whenever the stored devices for a user have been updated
+     * @param {string[]} userIds - A list of user IDs that were updated
+     * @param {boolean} initialFetch - If true, the store was empty (apart from our own device) and has been seeded.
+     */
+    DevicesUpdated = "crypto.devicesUpdated",
+    /**
+     * Fires when the user's cross-signing keys have changed or cross-signing
+     * has been enabled/disabled. The client can use getStoredCrossSigningForUser
+     * with the user ID of the logged in user to check if cross-signing is
+     * enabled on the account. If enabled, it can test whether the current key
+     * is trusted using with checkUserTrust with the user ID of the logged
+     * in user. The checkOwnCrossSigningTrust function may be used to reconcile
+     * the trust in the account key.
+     *
+     * The cross-signing API is currently UNSTABLE and may change without notice.
+     * @experimental
+     */
+    KeysChanged = "crossSigning.keysChanged",
+
+    /**
+     * Fires when data is being migrated from legacy crypto to rust crypto.
+     *
+     * The payload is a pair `(progress, total)`, where `progress` is the number of steps completed so far, and
+     * `total` is the total number of steps. When migration is complete, a final instance of the event is emitted, with
+     * `progress === total === -1`.
+     */
+    LegacyCryptoStoreMigrationProgress = "crypto.legacyCryptoStoreMigrationProgress",
+}
+
+export type CryptoEvents = (typeof CryptoEvent)[keyof typeof CryptoEvent];
```

Comment:
Again, if we're going to export this, please add a doc-comment explaining what it means.

Is it actually useful though? It doesn't seem like a thing we should be cluttering the js-sdk's API with.